htdocs/core/lib/security.lib.php

```diff
@@ -122,7 +122,7 @@ function dol_hash($chain, $type = '0')
 	}
 
 	// Salt value
-	if (! empty($conf->global->MAIN_SECURITY_SALT)) $chain=$conf->global->MAIN_SECURITY_SALT.$chain;
+	if (! empty($conf->global->MAIN_SECURITY_SALT) && ( ! $type == '4' || ! $type == 'md5openldap')) $chain=$conf->global->MAIN_SECURITY_SALT.$chain;
```

Comment:
! $type == '4'  means (! $type) == '4'
I don't think this is want you want to do.